src/token/token.js

```diff
@@ -40,18 +41,36 @@ class Token {
 		 * `create` method creates token with initialized value from url.
 		 *
 		 * @name value
-		 * @type {String}
+		 * @member {String} #value
 		 * @observable
 		 * @readonly
-		 * @memberOf Token#
 		 */
 		this.set( 'value', options.initValue );
 
 		/**
+		 * Refresh token function.
+		 *
+		 * @member {Function} #_refreshToken
+		 * @protected
+		 */
+		if ( typeof tokenUrlOrRefreshToken === 'function' ) {
+			this._refreshToken = () => {
+				return tokenUrlOrRefreshToken()
+					.then( value => this.set( 'value', value ) )
+					.then( () => this );
+			};
+		} else {
+			this._refreshToken = this._defaultRefreshToken.bind( this );
```

Comment:
I am not sure if `_defaultRefreshToken` should be still the method of this class. Maybe it could be just an internal function?

Reply:
I was thinking about it too. I'll move it totally outside.